src/workflow/confirm.h

```diff
@@ -28,6 +29,26 @@
  */
 bool workflow_confirm(const char* title, const char* body, bool longtouch, bool accept_only);
 
+/**
+ * Confirm something with the user asynchronously.
+ *
+ * Call this function with the same parameters until it returns WORKFLOW_ASYNC_READY. Then and only
```

Comment:
Slightly awkward that this is not enforced, and not technically required. I am guessing with one more enum entry we could have a request(params...) and poll(result) function. For later.

Reply:
In rust you could return `Ready(T)`...

Reply:
what does that mean? :)

Reply:
`Ready` could contain the return value.

```rust
enum Ready<T> {
    NotReady,
    Ready(T),
}
```

Reply:
The builtin type for async in rust is

```rust
/// Indicates whether a value is available or if the current task has been
/// scheduled to receive a wakeup instead.
#[must_use = "this `Poll` may be a `Pending` variant, which should be handled"]
#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
pub enum Poll<T> {
    /// Represents that a value is immediately ready.
    Ready(T),

    /// Represents that a value is not ready yet.
    ///
    /// When a function returns `Pending`, the function *must* also
    /// ensure that the current task is scheduled to be awoken when
    /// progress can be made.
    Pending,
}
```